sdk/keyvault/azure-security-keyvault-secrets/src/private/keyvault_protocol.hpp

```diff
@@ -185,5 +185,12 @@ namespace Azure { namespace Security { namespace KeyVault { namespace _detail {
       // Use the core pipeline directly to avoid checking the response code.
       return m_pipeline.Send(request, context);
     }
+
+    /**
+     * @brief Get the Url used to create the secret client.
+     *
+     * @return A constant reference to the Url.
+     */
+    Azure::Core::Url const& GetUrl() const { return m_vaultUrl; }
```

Comment:
Why does this one return `Azure::Core::Url` instead of `std::string`?

Reply:
Because it is in the implementation details. This class ( KeyVaultProtocol ) is used to create, send and parse all the requests from the convenience layer. It wraps the Url and Pipeline as private members so the secrets client only cares about one field.

In short, this is not a public client, but it's consumed by one.

Reply:
Is this actually needed? No one seems to be calling it. Can we remove it for now?

Reply:
Yes, it's used by the secret-client to produce the Url that is returned on its `GetUrl` method :)

Reply:
Eventually, @gearama  or someone will update the secrets api implementation to remove the protocol client (like we did with the keys-client) and this function will be removed along with all the protocol client :)

Reply:
Oh gotcha. Missed that. Thanks.